source/Hardpoint.h

```diff
@@ -40,13 +57,18 @@ class Hardpoint {
 	const Point &GetPoint() const;
 	// Get the angle that this weapon is aimed at, relative to the ship.
 	const Angle &GetAngle() const;
-	// Get the base angle that this weapon is aimed at (without harmonization/convergence), relative to the ship.
-	const Angle &GetBaseAngle() const;
+	// Get the angle of a turret when idling, relative to the ship.
+	// For guns, this function is equal to GetAngle().
+	const Angle &GetIdleAngle() const;
+	// Get the arc of fire if this is a directional turret,
+	// otherwise a pair of 180 degree + baseAngle.
+	const std::pair<Angle, Angle> &GetArc() const;
```

Comment:
I think it isn't too bad. I see about changing it if I have time (and hopefully don't mess anything up).

Reply:
Well trust me this isn't making it into the game this way; we can't use std::pair because things in it are not named. (and just called using .first, .second)
Nothing urgent either, of course, just something that will have to be changed at some point.

Reply:
Now that I fixed the major bug I'll try to change this, though I'm not sure what name should I give it. lowerLimit, upperLimit?

Reply:
More like minimumArc and maximumArc? Idk what the exact names should be tbh

Reply:
I'll go with minArc and maxArc.